ui/app/src/components/PreviewComponentsPanel/PreviewComponentsPanel.tsx

```diff
@@ -55,51 +57,63 @@ const translations = defineMessages({
   listInPageInstances: {
     id: 'previewComponentsPanel.listInPageInstances',
     defaultMessage: 'Instances on this page'
+  },
+  filter: {
+    id: 'previewComponentsPanel.filter',
+    defaultMessage: 'Filter...'
   }
 });
 
-type ComponentsPanelUIProps = PropsWithResource<ContentType[]>;
+type ComponentsPanelUIProps = { componentTypes: ContentType[] };
 
 export function PreviewComponentsPanel() {
-  const resource = useSelectorResource<ContentType[], EntityState<ContentType>>((state) => state.contentTypes, {
-    shouldRenew: (source, resource) => resource.complete,
-    shouldResolve: (source) => !source.isFetching && nnou(source.byId),
-    shouldReject: (source) => nnou(source.error),
-    errorSelector: (source) => source.error,
-    resultSelector: (source) =>
-      Object.values(reversePluckProps(source.byId, '/component/level-descriptor')).filter(
-        (contentType) => contentType.type === 'component'
+  const contentTypesBranch = useSelection((state) => state.contentTypes);
+  const [keyword, setKeyword] = useState('');
+  const contentTypes = contentTypesBranch.byId
+    ? Object.values(reversePluckProps(contentTypesBranch.byId, '/component/level-descriptor')).filter(
+        (contentType) =>
+          contentType.type === 'component' &&
+          (contentType.name.toLowerCase().includes(keyword.toLowerCase()) ||
+            contentType.id.toLowerCase().includes(keyword.toLowerCase()))
       )
-  });
+    : null;
+  const { formatMessage } = useIntl();
   return (
-    <>
-      <SuspenseWithEmptyState
-        resource={resource}
-        loadingStateProps={{
-          title: <FormattedMessage id="componentsPanel.suspenseStateMessage" defaultMessage="Retrieving Page Model" />
-        }}
-        withEmptyStateProps={{
-          emptyStateProps: {
-            title: <FormattedMessage id="componentsPanel.emptyStateMessage" defaultMessage="No components found" />,
-            subtitle: (
-              <FormattedMessage
-                id="componentsPanel.emptyComponentsSubtitle"
-                defaultMessage="Communicate with your developers to create the required components in the system."
-              />
-            )
+    <ErrorBoundary>
+      <Box sx={{ padding: (theme) => `${theme.spacing(1)} ${theme.spacing(1)} 0` }}>
+        <SearchBar
+          placeholder={formatMessage(translations.filter)}
+          showActionButton={Boolean(keyword)}
+          onChange={setKeyword}
+          keyword={keyword}
+          autoFocus
+        />
+      </Box>
+      {contentTypesBranch.isFetching ? (
+        <LoadingState
+          title={<FormattedMessage id="componentsPanel.suspenseStateMessage" defaultMessage="Retrieving Page Model" />}
+        />
+      ) : Object.keys(contentTypesBranch.byId).length === 0 ? (
+        <EmptyState
+          title={<FormattedMessage defaultMessage="No components found" />}
```

Comment:
Removing ids from existing translations would treat them as new. Leave existing ids.
```suggestion
          title={<FormattedMessage id="componentsPanel.emptyStateMessage" defaultMessage="No components found" />}
```